### PR TITLE
Added common files for lafaspot 

### PR DESCRIPTION
**Pull request by joshiaditya1993**

### CODE REVIEW

File: core/src/test/java/com/lafaspot/jmetrics/common/package-info.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  ====================================================================
+ */
+/**
+ * This package contains Unit Tests for classes in common.
+ *
+ */
+package com.lafaspot.jmetrics.common;
```
Comment (lafa) [inline core/src/test/java/com/lafaspot/jmetrics/common/package-info.java]: fix this

File: core/src/main/java/com/lafaspot/jmetrics/common/package-info.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *  ====================================================================
+ */
+
+
+/**
+ * MetricOutputter.
+ * 
+ * This package contains the code common to several classes.
+ */
+package com.lafaspot.jmetrics.common;
```
Comment (lafa) [inline core/src/main/java/com/lafaspot/jmetrics/common/package-info.java]: new line please

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitor.java [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+import com.lafaspot.jmetrics.annotation.MetricCheck;
+import com.lafaspot.jmetrics.annotation.MetricClass;
+import com.lafaspot.jmetrics.datatype.TimeValue;
+
+/**
+ * Records incoming http request stats in an interval.
+ *
+ */
+@MetricClass(enable = true, name = "ContainerMonitor", applications = { "admin" })
+public class ContainerMonitor implements ContainerMonitorMBean {
```
Comment (lafa) [inline jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitor.java]: change the bean and the monitor class to a jmx_sample package

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitorMBean.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ *  ====================================================================
+ */
+
+package com.lafaspot.jmetrics.monitor;
+
+/**
+ * Container monitor interface.
+ *
+ */
+public interface ContainerMonitorMBean extends BaseMonitor {
```
Comment (lafa) [inline jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitorMBean.java]: move this to jmx_sample package

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/package-info.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  ====================================================================
+ */
+
+/**
+ * This package has code related to monitors.
+ *
+ */
+package com.lafaspot.jmetrics.monitor;
```
Comment (lafa) [inline jmx/src/main/java/com/lafaspot/jmetrics/monitor/package-info.java]: add new line

File: jmx/src/test/java/com/lafaspot/jmetrics/common/ContainerMonitorTest.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+
+/**
+ * Test class for ContainerMonitor.
+ *
+ */
+public class ContainerMonitorTest {
```
Comment (lafa) [inline jmx/src/test/java/com/lafaspot/jmetrics/common/ContainerMonitorTest.java]: more to jmx_sample pacakge

File: jmx/src/test/java/com/lafaspot/jmetrics/common/TimeValueTest.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+import com.lafaspot.jmetrics.datatype.TimeValue;
+
+/**
+ * Unit tests of the {@linkplain TimeValue}.
+ * 
+ * @author manish211
+ *
+ */
+public class TimeValueTest {
```
Comment (lafa) [inline jmx/src/test/java/com/lafaspot/jmetrics/common/TimeValueTest.java]: move to common

File: jmx/src/main/java/com/lafaspot/jmetrics/datatype/TimeValue.java [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+
+import javax.annotation.Nonnull;
+
+/**
+ * Physical quantities with units of time, and 64-bit integer values.
+ *
+ * @author manish211
+ *
+ */
+public final class TimeValue implements java.io.Serializable, Comparable<TimeValue> {
```
Comment (lafa) [inline jmx/src/main/java/com/lafaspot/jmetrics/datatype/TimeValue.java]: move to common.datatype

**Comment by squintview** [comment]
Need to move the datatypes to lafaspot/common - something like - /master/core/src/main/java/com/lafaspot/common/types

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitor.java (hunk 3 above)
Comment (joshiaditya1993) [reply jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitor.java]: Done

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitorMBean.java (hunk 4 above)
Comment (joshiaditya1993) [reply jmx/src/main/java/com/lafaspot/jmetrics/monitor/ContainerMonitorMBean.java]: Done

File: jmx/src/main/java/com/lafaspot/jmetrics/monitor/package-info.java (hunk 5 above)
Comment (joshiaditya1993) [reply jmx/src/main/java/com/lafaspot/jmetrics/monitor/package-info.java]: Done

File: jmx/src/test/java/com/lafaspot/jmetrics/common/ContainerMonitorTest.java (hunk 6 above)
Comment (joshiaditya1993) [reply jmx/src/test/java/com/lafaspot/jmetrics/common/ContainerMonitorTest.java]: Done

File: jmx/src/main/java/com/lafaspot/jmetrics/datatype/TimeValue.java (hunk 8 above)
Comment (joshiaditya1993) [reply jmx/src/main/java/com/lafaspot/jmetrics/datatype/TimeValue.java]: Done

File: jmx/src/test/java/com/lafaspot/jmetrics/common/TimeValueTest.java (hunk 7 above)
Comment (joshiaditya1993) [reply jmx/src/test/java/com/lafaspot/jmetrics/common/TimeValueTest.java]: Done

**Comment by lafa** [comment]
still missing a seperate project form jmx_sample classes, please create a new maven project called jmx_sample, and move the jmx_sample package there, so that the clients that use the jmx packages dont't get the sample classes.

**Comment by lafa** [comment]
please also cancel the other PR, that seems co contain the same code changes.

**Comment by manish211** [comment]
closed my previous pull request.

**Pull request merged**